contentcuration/contentcuration/frontend/shared/vuex/indexedDBPlugin/index.js

```diff
@@ -0,0 +1,161 @@
+import { EventEmitter } from 'events';
```

Comment:
I feel like the event emitter + listener stuff is adding an extra layer of indirection here that I don't quite see the impetus for, but it's also cleanly written, tested code, so this is not a blocker.

Reply:
The addition of EventEmitter allowed for removing code that needed unit tests, so since it does the same thing, it was a no brainer. Also, dealing with an object like an event emitter instance or Listener, makes things easier to test by encapsulating functionality thereby making it easier to mock, and by breaking it down its functionality into isolate-able chunks for individual unit tests. Also, in my opinion it makes important code flows easier to read by reducing it to objects and their operations, simplifying the amount of state that has to be managed. `Listener` and `ListenerGroup` actually follow the Composite pattern, at least until I delete `ListenerGroup`, a very useful OOP pattern.

Reply:
I think it's just that the Listener is used twice, so it feels like an early abstraction, but I get that it helps with testing. Not a blocker by any means.